sdk/face/ai-vision-face-rest/src/models.ts

```diff
@@ -0,0 +1,144 @@
+  deviceCorrelationId?: string;
+  /** Seconds the session should last for. Range is 60 to 86400 seconds. Default value is 600. */
+  authTokenTimeToLiveInSeconds?: number;
+}
+
+export interface ICreateLivenessWithVerifySessionContentParametersPartDescriptor {
+  name: "Parameters";
+  body: CreateLivenessSessionContentForMultipart;
+}
+
+export class CreateLivenessWithVerifySessionContentParametersPartDescriptor {
+  get name(): string {
+    return "Parameters";
+  }
+
+  body: CreateLivenessSessionContentForMultipart;
+
+  constructor(descriptor: ICreateLivenessWithVerifySessionContentParametersPartDescriptor) {
+    this.body = descriptor.body;
+  }
+}
+
+export interface ICreateLivenessWithVerifySessionContentVerifyImagePartDescriptor {
+  name: "VerifyImage";
+  body:
+    | string
+    | Uint8Array
+    | ReadableStream<Uint8Array>
+    | NodeJS.ReadableStream
+    | File;
+  filename?: string;
+  contentType?: string;
+}
+
+export class CreateLivenessWithVerifySessionContentVerifyImagePartDescriptor {
+  get name(): string {
+    return "VerifyImage";
+  }
+
+  get contentType(): string {
+    return "application/octet-stream";
+  }
+
+  body: string | Uint8Array | ReadableStream<Uint8Array> | NodeJS.ReadableStream | File;
+
+  filename: string;
+
+  constructor(descriptor: ICreateLivenessWithVerifySessionContentVerifyImagePartDescriptor) {
+    this.body = descriptor.body;
+    this.filename = descriptor.filename ?? "verify-image";
+  }
+}
+
+/** Dedicated parameter model for multipart/form-data. */
+export interface CreateLivenessSessionContentForMultipart {
+  /** Type of liveness mode the client should follow. */
+  livenessOperationMode: LivenessOperationMode;
+  /** Whether or not to allow a '200 - Success' response body to be sent to the client, which may be undesirable for security reasons. Default is false, clients will receive a '204 - NoContent' empty body response. Regardless of selection, calling Session GetResult will always contain a response body enabling business logic to be implemented. */
+  sendResultsToClient?: boolean;
+  /** Whether or not to allow client to set their own 'deviceCorrelationId' via the Vision SDK. Default is false, and 'deviceCorrelationId' must be set in this request body. */
+  deviceCorrelationIdSetInClient?: boolean;
+  /** Unique Guid per each end-user device. This is to provide rate limiting and anti-hammering. If 'deviceCorrelationIdSetInClient' is true in this request, this 'deviceCorrelationId' must be null. */
+  deviceCorrelationId?: string;
+  /** Seconds the session should last for. Range is 60 to 86400 seconds. Default value is 600. */
+  authTokenTimeToLiveInSeconds?: number;
+}
+
+/** Alias for DetectionModel */
+export type DetectionModel =
+  | string
+  | "detection_01"
+  | "detection_02"
+  | "detection_03";
```

Comment:
if the new extensible enum design is approved, will it be a breaking change? @joheredi

Reply:
Probably, it depends what the intention of this is. If this means "This is a closed set for this version but more may be added next version" yes, it will be a breaking change. If this means "This can be any string, but here's a set of known values that can be used", it won't be a breaking change